File: packages/python/tests/integration/__fixtures__/dirsql.config.py
"""A native Python config file for dirsql.

When ``dirsql --config <path-to-this-file>`` runs, the Rust binary
spawns ``dirsql interpret`` against this file. The helper loads the
module, takes its module-level ``app``, and dispatches ``extract``
callbacks over NDJSON.
"""

import json
import os

from dirsql import DirSQL, Table


def _extract(path):
    with open(path) as f:
        return [json.load(f)]


app = DirSQL(
    root=os.path.join(os.path.dirname(__file__), "data"),
    tables=[
        Table(
            ddl="CREATE TABLE papers (title TEXT)",
            glob="**/meta.json",
            extract=_extract,
        )
    ],
)

File: packages/ts/tests/e2e/__fixtures__/dirsql.config.js
// A native .js config file for dirsql. The launcher resolves .js as
// ESM or CJS based on the nearest package.json's `type` field; here the
// fixture sits inside an ESM package and is parsed as ESM.
//
// When `dirsql --config <path-to-this-file>` runs, the Rust binary
// spawns `dirsql interpret` against this file. The helper loads the
// module, takes its default export, and dispatches `extract` callbacks
// over NDJSON.

import { readFileSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";
import { DirSQL } from "dirsql";

const here = dirname(fileURLToPath(import.meta.url));

export default new DirSQL({
  root: join(here, "data"),
  tables: [
    {
      ddl: "CREATE TABLE papers (title TEXT)",
      glob: "**/meta.json",
      extract: (path) => [JSON.parse(readFileSync(path, "utf8"))],
    },
  ],
});
